### PR TITLE
Feature/co1535 implement integration test

### PR DESCRIPTION
**Pull request by nosnatef**

### CODE REVIEW

File: tests/integration/configuration-example.json [hunk 1]
```diff
@@ -1,22 +0,0 @@
-{
```
Comment (Ruefa) [inline tests/integration/configuration-example.json]: This file is needed for others to run your tests. See [readme](https://github.com/osu-mist/poker-api/blob/feature/CO1535-implement-integration-test/tests/integration/README.md#test-on-local-instance). It also needs to be setup to reflect the data needed in your tests. [Example](https://github.com/osu-mist/game-curator-api/blob/4929a92be94da94b9fb2084382360c9832da033b/tests/integration/configuration-example.json#L18-L37)

File: api/v1/serializers/games-serializer.js [hunk 2]
```diff
@@ -52,13 +53,17 @@ const mergeRawGames = (rawGames) => {
...
 
-const serializeGames = (rawGames, query) => {
+const serializeGames = (rawGames, query, memberId) => {
   rawGames = mergeRawGames(rawGames);
   _.forEach(rawGames, (game) => {
     individualGameConverter(game);
   });
+  const gameResourcePathInstance = memberId ? gameFromMemberResourcePath(memberId)
+    : gameResourcePath;
+  console.log(gameResourcePathInstance);
```
Comment (Ruefa) [inline api/v1/serializers/games-serializer.js]: Remove debug statement

File: tests/integration/integration-test.py [hunk 3]
```diff
@@ -31,76 +31,240 @@ def setup(cls, config_path, openapi_path):
         cls.openapi = parser.specification
 
     @classmethod
-    def cleanup(cls):
```
Comment (Ruefa) [inline tests/integration/integration-test.py:34]: Why did you rename and remove the reference to this class?

**Review by Ruefa — CHANGES_REQUESTED** [review]
Can't run the tests until configuration-example.json is setup. Also one of the unit tests is failing. The integration tests themselves look great.

File: tests/integration/integration-test.py (hunk 3 above)
Comment (nosnatef) [reply tests/integration/integration-test.py:34]: We had a bug and Ian told me to delete this function.

File: api/v1/paths/members/{memberId}/games.js [hunk 4]
```diff
@@ -3,7 +3,6 @@ const appRoot = require('app-root-path');
 const gameDao = require('../../../db/oracledb/game-dao');
 
 const { errorBuilder, errorHandler } = appRoot.require('errors/errors');
-const { openapi: { paths } } = appRoot.require('utils/load-openapi');
```
Comment (Ruefa) [inline api/v1/paths/members/{memberId}/games.js:6]: Why'd you remove this line? The api crashes without it.

**Review by Ruefa — CHANGES_REQUESTED** [review]
I tried to seed MEMBER data using post and hit this `500` error `2019-08-22T17:11:10.892Z - error: Error: ORA-00904: "M"."MEMBER_NICKNAME": invalid identifier`. Could you fix this and seed some data for me to run the tests?

File: api/v1/paths/members/{memberId}/games.js (hunk 4 above)
Comment (nosnatef) [reply api/v1/paths/members/{memberId}/games.js:6]: I tried to pull the changes from skeleton to the API, and this line is not required. I merged another branch to this branch so the problem can be fixed.

File: package.json [hunk 5]
```diff
@@ -34,33 +35,35 @@
...
-    "express-winston": "^3.0.0",
+    "express-winston": "^3.2.1",
     "js-yaml": "^3.13.1",
     "jsonapi-serializer": "^3.5.6",
     "lodash": "^4.17.14",
     "moment": "^2.22.2",
     "oracledb": "^3.1.2",
     "query-string": "^6.2.0",
     "simple-git": "^1.96.0",
+    "sinon": "^7.3.2",
```
Comment (ianbrown9475) [inline package.json:46]: Could you move `sinon` back to `devDependencies`?

**Review by ianbrown9475 — APPROVED** [review]

**Review by Ruefa — CHANGES_REQUESTED** [review]
I still had the same error when trying to post after pulling the changes.

I inserted the following data based on config-example:
```
insert into members (MEMBER_ID, MEMBER_NAME, MEMBER_EMAIL, MEMBER_LEVEL, MEMBER_EXP_OVER_LEVEL, MEMBER_PASSWORD) values (1, 'J', 'abc@efg.com', 0, 0, 'nunya');
insert into members (MEMBER_ID, MEMBER_NAME, MEMBER_EMAIL, MEMBER_LEVEL, MEMBER_EXP_OVER_LEVEL, MEMBER_PASSWORD) values (2, 'John Wick', 'wickj@oregonstate.edu', 0, 0, 'nunya');
insert into members (MEMBER_ID, MEMBER_NAME, MEMBER_EMAIL, MEMBER_LEVEL, MEMBER_EXP_OVER_LEVEL, MEMBER_PASSWORD) values (3, 'patchedNickname', 'patchedEmail@gmail.com', 0, 0, 'nunya');

insert into games (GAME_ID, ROUND_ID, MINIMUM_BET, MAXIMUM_BET, BET_POOL) values (1, 1, 1, 10, 2);
insert into games (GAME_ID, ROUND_ID, MINIMUM_BET, MAXIMUM_BET, BET_POOL) values (2, 1, 1, 10, 2);
insert into games (GAME_ID, ROUND_ID, MINIMUM_BET, MAXIMUM_BET, BET_POOL) values (3, 1, 1, 10, 2);

insert into players (PLAYER_ID, MEMBER_ID, PLAYER_BET, STATUS_ID) values (1, 1, 1, 1);
insert into players (PLAYER_ID, MEMBER_ID, PLAYER_BET, STATUS_ID) values (2, 2, 1, 2);
insert into players (PLAYER_ID, MEMBER_ID, PLAYER_BET, STATUS_ID) values (3, 3, 1, 3);
```

With this data I had 30 tests failures. You may need to add to the [README](https://github.com/osu-mist/poker-api/blob/master/tests/integration/README.md) information on what data needs to be inserted to run the tests.

**Comment by nosnatef** [comment]
Did you use the example post body from the swagger file? I opened the swagger file in the swagger editor and just used them to test my API. Everything works just fine. @Ruefa

**Comment by nosnatef** [comment]
Ian and I confirmed that it was the problem of wrong database Alex was using.

**Review by Ruefa — APPROVED** [review]
